Tests/ZIPFoundationTests/ZIPFoundationWritingTests+Zip64.swift

```diff
@@ -0,0 +1,316 @@
+                try archive.addFileEntry(with: entryName, size: size, data: data)
+            }
+        } catch {
+            XCTFail("Failed to add zip64 format entry to archive with error : \(error)"); return
+        }
+        XCTAssertEqual(archive.endOfCentralDirectoryRecord.totalNumberOfEntriesInCentralDirectory, UInt16(factor - 1))
+        XCTAssertNil(archive.zip64EndOfCentralDirectory?.record.totalNumberOfEntriesInCentralDirectory)
+        // Case 2: The total number os entries is equal to maximum value
+        do {
+            try archive.addEntry(with: "Test", type: .directory,
+                                 uncompressedSize: 0, provider: { _, _ in return Data() })
+        } catch {
+            XCTFail("Failed to add zip64 format entry to archive with error : \(error)"); return
+        }
+        XCTAssertEqual(archive.endOfCentralDirectoryRecord.totalNumberOfEntriesInCentralDirectory, UInt16.max)
+        XCTAssertEqual(archive.zip64EndOfCentralDirectory?.record.totalNumberOfEntriesInCentralDirectory ?? 0,
+                       UInt(factor))
+    }
+
+    /// Target fields: Size of Central Directory
+    func testCreateZip64ArchiveWithLargeSizeOfCD() {
+        let factor = 10
+        mockIntMaxValues(int32Factor: factor)
+        defer { resetIntMaxValues() }
+        let archive = self.archive(for: #function, mode: .create)
+        let size = 64
+        // Case 1: The size of central directory is less than maximum value
+        do {
+            try archive.addEntry(with: "link", type: .symlink, uncompressedSize: size,
+                                 provider: { (_, count) -> Data in
+                                    return Data(count: count)
+            })
+        } catch {
+            XCTFail("Failed to add zip64 format entry to archive with error : \(error)"); return
+        }
+        XCTAssertLessThan(archive.endOfCentralDirectoryRecord.sizeOfCentralDirectory, UInt32.max)
+        XCTAssertNil(archive.zip64EndOfCentralDirectory?.record.sizeOfCentralDirectory)
+        // Case 2: The size of central directory is greater than maximum value
+        do {
+            let data = Data.makeRandomData(size: size)
+            let entryName = ProcessInfo.processInfo.globallyUniqueString
+            try archive.addFileEntry(with: entryName, size: size, data: data)
+        } catch {
+            XCTFail("Failed to add zip64 format entry to archive with error : \(error)"); return
+        }
+        XCTAssertEqual(archive.endOfCentralDirectoryRecord.sizeOfCentralDirectory, UInt32.max)
+        XCTAssertLessThan(0, archive.zip64EndOfCentralDirectory?.record.sizeOfCentralDirectory ?? 0)
+    }
+
+    func testRemoveEntryFromArchiveWithZip64EOCD() {
+        /*
+         File structure:
+         testRemoveEntryFromArchiveWithZip64EOCD.zip/
+           ├─ data1.random (size: 64)
+           ├─ data2.random (size: 64 * 64)
+         */
```

Comment:
Great comment!
Nit: Would use `//` single line comment syntax for all lines. (Instead of `/* */`)